core/arch/arm/mm/core_mmu_v7.c

```diff
@@ -717,29 +717,24 @@ static paddr_t map_page_memarea(const struct tee_mmap_region *mm, uint32_t xlat)
 	return virt_to_phys(l2);
 }
 
-static void map_page_memarea_in_pgdirs(const struct tee_mmap_region *mm,
+static void map_page_memarea_in_pgdir(const struct tee_mmap_region *mm,
 					uint32_t *ttb)
 {
-	uint32_t attr = INVALID_DESC;
+	uint32_t attr;
 	size_t idx = mm->va >> SECTION_SHIFT;
-	paddr_t pa = 0;
-	size_t n;
+	paddr_t pa;
 
 	if (core_mmap_is_end_of_table(mm))
 		return;
 
 	print_mmap_area(mm, "4k page map");
 
 	attr = mattr_to_desc(1, mm->attr | TEE_MATTR_TABLE);
+	assert(idx == ((mm->va + mm->size) >> SECTION_SHIFT));
```

Comment:
My proposal was buggy: is `mm->va + mm->size` reaches the base address of the next section, the map request is still valid but current assert will fail.
Fix proposal:
```patch
-	assert(idx == ((mm->va + mm->size) >> SECTION_SHIFT));
+	assert(mm->size && idx == ((mm->va + mm->size - 1) >> SECTION_SHIFT));

Reply:
Yes, you are right. Code is corrected.